### DISCUSSION TITLE
A top-level wrapper command for convenience

### DISCUSSION THREAD

**Discussion by ossdhaval** [JanssenProject/jans#8616] [Ideas]
The most frequent commands used by Janssen Server users/admins involve commands which needs users to either remember the path or parameters. This can be greatly simplified if we add few wrapper commands which are easy to invoke.

For instance, 
- to invoke jans-cli, the admin has to remember where the script is: `/opt/jans/jans-cli/config-cli.py`. It would be a lot simpler to invoke something like `jans cli`
- to get the status of services, the command is `sudo systemctl list-units --all "jans*"`. Instead, it is much simpler to invoke `jans status`

I propose to add a `jans` command as a top level command using which other subcommand can be easily invoked. 

For example:

Simply invoking `jans` would show help content like below:

```
Usage:  jans [OPTIONS] COMMAND

Common Commands:
  status      Show status of Janssen Server module services
  cli            invokes jans-cli
  tui           invokes jans text based user interface
  version   shows version of currently installed Janssen Server
  logs        Shows the log file paths for various Janssen Server modules
  start        Starts services for Janssen Server ( -service=<service-name> if you want to start just one specific service)
  stop        Stops services for Janssen Server ( -service=<service-name> if you want to stop just one specific service)
  restart     restarts services for Janssen Server ( -service=<service-name> if you want to restart just one specific service)
  health     gets output from Janssen `health-check` endpoint
  info         lists important URLs (like .well-known, Casa etc)
```

`jans` command basically makes information available on users finger tips. And all they have to remember is one command `jans`.

What do you think?

**Comment by manojs1978** [comment]
it will be good if we have such shorts/commands  from users perspective.

**Reply by devrimyatar** [reply]
I like the idea. May be other developers can help extending the list.

**Comment by moabu** [comment]
The wrapper is meant to be for the linux based installation only.

**Reply by ossdhaval** [reply]
@moabu You mean it'll be only useful on `vm` installs and not in cloud native?

**Reply by moabu** [reply]
yes. This has no meaning in Cloud installations. If people install using kubernetes they already have kubectl. Plus this is more suited for a one setup structure.

**Comment by ossdhaval** [comment]
Hi @devrimyatar I think we all agree that this feature is going to be useful if implemented. I am creating an issue to track it.

**Comment by ossdhaval** [comment]
Thanks @devrimyatar for [contributing this feature](https://github.com/JanssenProject/jans/issues/8697) :clap: 

It is lovely :heart: to see the improved user experience on commandline. On the Janssen Server, just type `jans` on the command line and hit enter to see the available magic :sparkles: options.

I am closing this discussion now. All the suggestions for additional options for `jans` command should be proposed through feature request issues.
